### PR TITLE
Added unit tests for MultiModelEngine

### PR DESCRIPTION
**Pull request by SageMoore**
These tests take about 90 seconds to  run on my 18 core computer.

### CODE REVIEW

**Comment by dbarbuzzi** [comment]
Worth noting that the GitHub action base test check with all of the added tests took just over 13m to run (previously this check was probably 5-7m on average). If this is too long, some or all could be removed by removing the smoke tag from them. Partial coverage would make it a bit messy as the loops would have to be changed/unrolled to only tag certain parametrizations as smoke.

**Comment by SageMoore** [comment]
> 

This is definitely too much. I'll break some of the tests out.

**Comment by SageMoore** [comment]
@dbarbuzzi I made a minor fix to the model selection logic and it looks like that dropped the time to 7 minutes. Is that OK?

**Comment by dbarbuzzi** [comment]
Yeah, 7 minutes should be fine; my given target when setting up the GitHub actions was less than 10 minutes.

File: tests/test_multi_engine.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        pred_outputs = engine(inputs)
+        verify_outputs(pred_outputs, outputs)
+
+    def test_multi_engine(
+        self,
+        num_streams: int,
+        num_requests: int,
+    ):
+        models = list(model_test_registry.values())
+        batch_size = 64
```
Comment (mgoin) [inline tests/test_multi_engine.py:59]: do we need a large batch size here?

Comment (SageMoore) [reply tests/test_multi_engine.py]: The batch size doesn't make any difference in terms of coverage. I just picked 64. 1 would also be fine. Most of the runtime of the test is dominated by compilation so I don't have a strong opinion here. On my computer b1 is about 5 seconds faster (out of 75). The runtime difference could be more significant on the github action vm so we can use 1 instead and see if it helps.

**Comment by SageMoore** [comment]
Ok the first time I used b1 the total runtime of the base-tests job was 10 minutes. The second time it was 6. I assume this kind of variance is somewhat expected? Both times I ran the b64 job it took 7 minutes.

**Comment by dbarbuzzi** [comment]
In the short time these tests have been running, I have seen a range of ~4-7 minutes so 6-10 is not too surprising.

We can consider adding a flag to the runs which would include per-test runtime in the output if we wanted to track that. It wouldn’t otherwise affect the runs’ outcome and would only be visible if you drilled deep into the actions to see the specific step’s output.

**Comment by SageMoore** [comment]
It's worth mentioning that on the 10 minute run, the tests that normally take 4 minutes were taking 7 so it could have just been a slow run overall.

**Review by mgoin — APPROVED** [review]

**Pull request merged**